app/models/participant_profile_state.rb

```diff
@@ -1,9 +1,26 @@
 # frozen_string_literal: true
 
 class ParticipantProfileState < ApplicationRecord
+  def valid?(_context = nil)
+    current_state = participant_profile.participant_profile_state
+    return true if current_state.nil?
+
+    case state
+    when "active"
+      errors.add(:base, "already active") if current_state.active?
+    when "withdrawn"
+      errors.add(:base, I18n.t(:invalid_withdrawal)) if current_state.withdrawn?
+    when "deferred"
+      errors.add(:base, I18n.t(:invalid_withdrawal)) if current_state.withdrawn?
```

Comment:
I wonder if those two should be the same error messages. 
> The participant is already withdrawn

sounds good when you try to withdraw, perhaps slightly less good when you try to defer.